ArchiSteamFarm/IPC/Startup.cs

```diff
@@ -97,44 +98,47 @@ internal sealed class Startup {
 		// Add support for default root path redirection (GET / -> GET /index.html), must come before static files
 		app.UseDefaultFiles();
 
-		// Add support for static files (e.g. HTML, CSS and JS from IPC GUI)
-		app.UseStaticFiles(
-			new StaticFileOptions {
-				OnPrepareResponse = static context => {
-					if (context.File is { Exists: true, IsDirectory: false } && !string.IsNullOrEmpty(context.File.Name)) {
-						string extension = Path.GetExtension(context.File.Name);
-
-						CacheControlHeaderValue cacheControl = new();
-
-						switch (extension.ToUpperInvariant()) {
-							case ".CSS" or ".JS":
-								// Add support for SRI-protected static files
-								// SRI requires from us to notify the caller (especially proxy) to avoid modifying the data
-								cacheControl.NoTransform = true;
-
-								goto default;
-							default:
-								// Instruct the caller to always ask us first about every file it requests
-								// Contrary to the name, this doesn't prevent client from caching, but rather informs it that it must verify with us first that his cache is still up-to-date
-								// This is used to handle ASF and user updates to WWW root, we don't want from the client to ever use outdated scripts
-								cacheControl.NoCache = true;
-
-								// All static files are public by definition, we don't have any authorization here
-								cacheControl.Public = true;
-
-								break;
-						}
+#if !NETFRAMEWORK && !NETSTANDARD
+		List<string> staticFilesDirectorys = new();
 
-						ResponseHeaders headers = context.Context.Response.GetTypedHeaders();
+		string pluginsPath = Path.Combine(SharedInfo.HomeDirectory, SharedInfo.PluginsDirectory);
 
-						headers.CacheControl = cacheControl;
-					}
-				}
+		if (Directory.Exists(pluginsPath)) {
+			List<string>? staticFilesDirs = GetPluginsStaticFilesPathFrom(pluginsPath);
+
+			if (staticFilesDirs?.Count > 0) {
+				staticFilesDirectorys.AddRange(staticFilesDirs);
 			}
-		);
+		}
+
+		string customPluginsPath = Path.Combine(Directory.GetCurrentDirectory(), SharedInfo.PluginsDirectory);
+
+		if ((pluginsPath != customPluginsPath) && Directory.Exists(customPluginsPath)) {
+			List<string>? staticFilesDirs = GetPluginsStaticFilesPathFrom(customPluginsPath);
+
+			if (staticFilesDirs?.Count > 0) {
+				staticFilesDirectorys.AddRange(staticFilesDirs);
+			}
+		}
+
+		foreach (string staticFilesDirectory in staticFilesDirectorys) {
+			app.UseDefaultFiles("/plugins/" + Directory.GetParent(staticFilesDirectory)?.Name);
```

Comment:
`Directory.GetParent(string)` will only return `null` if the parameter a root directory according to the [documentation](https://learn.microsoft.com/en-us/dotnet/api/system.io.directory.getparent?view=net-7.0#returns). Thus we should be able to use the `!` operator instead of `?`.

```suggestion
			app.UseDefaultFiles("/plugins/" + Directory.GetParent(staticFilesDirectory)!.Name);
```